src/lib/integration_test_lib/malleable_error.ml

```diff
@@ -0,0 +1,345 @@
+
+let return_unit_without_deferred =
+  Ok {Accumulator.computation_result= (); soft_errors= []}
+
+let ok_unit = return ()
+
+let ok_exn (res : 'a t) : 'a Deferred.t =
+  let open Deferred.Let_syntax in
+  match%bind res with
+  | Ok {Accumulator.computation_result= x; soft_errors= _} ->
+      Deferred.return x
+  | Error {Hard_fail.hard_error= err; Hard_fail.soft_errors= _} ->
+      Error.raise err.error
+
+let error_string message =
+  Deferred.return
+    (Error
+       { Hard_fail.hard_error=
+           Test_error.raw_internal_error (Error.of_string message)
+       ; soft_errors= [] })
+
+let errorf format = Printf.ksprintf error_string format
+
+let soft_error res err =
+  Deferred.return
+    (Ok
+       { Accumulator.computation_result= res
+       ; soft_errors= [Test_error.raw_internal_error err] })
+
+let hard_error err =
+  Deferred.return
+    (Error
+       { Hard_fail.hard_error= Test_error.raw_internal_error err
+       ; soft_errors= [] })
+
+let or_error_to_hard_malleable_error (or_err : 'a Deferred.Or_error.t) : 'a t =
+  let open Deferred.Let_syntax in
+  match%map or_err with
+  | Ok x ->
+      Ok {Accumulator.computation_result= x; soft_errors= []}
+  | Error err ->
+      Error
+        { Hard_fail.hard_error= Test_error.raw_internal_error err
+        ; soft_errors= [] }
+
+let or_error_to_soft_malleable_error (res : 'a)
+    (or_err : 'a Deferred.Or_error.t) : 'a t =
+  let open Deferred.Let_syntax in
+  match%map or_err with
+  | Ok x ->
+      Ok {Accumulator.computation_result= x; soft_errors= []}
+  | Error err ->
+      Ok
+        { Accumulator.computation_result= res
+        ; soft_errors= [Test_error.raw_internal_error err] }
+
+(* 
+'a Malleable_error.t list -> 'a list Malleable_error.t
+ *)
+(* let combine_errors (error_list : 'a Malleable_error.t list ) : 'a list Malleable_error.t = 
+  Deferred.map (Deferred.all error_list) ~f:(combine_errors_helper)
+
+let combine_errors_helper (error_list : ('a Accumulator.t, Hard_fail.t) Result.t list ) : (('a list Accumulator.t, Hard_fail.t) Result.t) =
+  List.fold_left error_list ~init:(return_without_deferred []) ~f:combine_errors_helper3
+
+let combine_errors_helper3 (fold_acc : ('a list Accumulator.t, Hard_fail.t) Result.t ) (mall_err: 'a Malleable_error.t) = *)
+
+let combine_errors (malleable_errors : 'a t list) : 'a list t =
+  let open T.Let_syntax in
+  let%map values =
+    List.fold_left malleable_errors ~init:(return []) ~f:(fun acc el ->
+        let%bind t = acc in
+        let%map h = el in
+        h :: t )
+  in
+  List.rev values
+
+let try_with ?(backtrace = false) (f : unit -> 'a) : 'a t =
+  try
+    Deferred.return
+      (Ok {Accumulator.computation_result= f (); soft_errors= []})
+  with exn ->
+    Deferred.return
+      (Error
+         { Hard_fail.hard_error=
+             Test_error.raw_internal_error
+               (Error.of_exn exn
+                  ?backtrace:(if backtrace then Some `Get else None))
```

Comment:
This will get the backtrace at the current line of code. This function will likely return better stacktraces if you use the existing `Or_error.try_with` function instead of doing your own try catch. Example:

```ocaml
let try_with (type a) ?(backtrace = false) (f : unit -> a) : a t =
  of_hard_or_error (Or_error.try_with ?backtrace ~f)
```